common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -241,28 +242,39 @@ def is_block_complete_for_assignments(self, completion_service):
         Compare with is_block_structure_complete_for_assignments in course_experience/utils.py, which does the same
         calculation, but for a BlockStructure node and its children.
 
-        Returns:
-            True if complete
-            False if not
-            None if no assignments present or no completion info present (don't show any past-due or complete info)
+        This method will also return a boolean on if the vertical has graded and scored children. This is
+        used to determine if we should show a due date in the courseware. We set it here so we only
+        consider completeable children.
+
+        Returns tuple of:
+            First value (completeness):
+                True if complete
+                False if not
+                None if no assignments present or no completion info present (don't show any past-due or complete info)
+            Second value (has graded and scored children):
+                True if has graded and scored children
+                False if not
+                False if no assignments present or no completion info present
         """
         if not completion_service or not completion_service.completion_tracking_enabled():
-            return None
+            return None, False
 
         children = completion_service.get_completable_children(self)
         children_locations = [child.scope_ids.usage_id for child in children]
         completions = completion_service.get_completions(children_locations)
 
         all_complete = None
+        has_graded_and_scored_children = False
         for child in children:
             complete = completions[child.scope_ids.usage_id] == 1
             graded = getattr(child, 'graded', False)
             has_score = getattr(child, 'has_score', False)
             weight = getattr(child, 'weight', 1)
             scored = has_score and (weight is None or weight > 0)
             if graded and scored:
+                has_graded_and_scored_children = True
                 if not complete:
-                    return False
+                    return False, has_graded_and_scored_children
                 all_complete = True
 
-        return all_complete
+        return all_complete, has_graded_and_scored_children
```

Comment:
`has_graded_and_scored_children` is always equal to `all_complete is not None` -- i.e. when the docstring says None means there are no assignments. `has_graded_and_scored_children` is just asking if there are assignments.

I'd vote to drop the extra complexity and just check for None. Eh?